apps/remote_control/lib/lexical/remote_control/code_intelligence/definition.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.Definition do
+  alias Lexical.SourceFile
+  alias Lexical.SourceFile.Position
+  alias Lexical.SourceFile.Range
+
+  require Logger
+
+  def definition(%SourceFile{} = source_file, %Position{} = position) do
+    maybe_location =
+      source_file
+      |> SourceFile.to_string()
+      |> ElixirSense.definition(position.line + 1, position.character + 1)
+
+    parse_location(maybe_location, source_file)
+  end
+
+  defp parse_location(%ElixirSense.Location{} = location, source_file) do
+    %{file: file, line: line, column: column} = location
+    file_path = file || source_file.path
+
+    case location_to_range(file_path, line, column) do
+      {:ok, {source_file, range}} ->
+        {:ok, %{source_file: source_file, range: range}}
+
+      error ->
+        error
+    end
+  end
+
+  defp parse_location(nil, _) do
+    {:ok, nil}
+  end
+
+  defp location_to_range(file_path, line, column) do
+    uri = SourceFile.Path.ensure_uri(file_path)
+
+    with {:ok, source_file} <- SourceFile.Store.open_temporary(uri),
+         {:ok, {:line, text, _, _, _}} <- SourceFile.fetch_line_at(source_file, line - 1) do
+      {:ok, {source_file, to_persisted_range(text, line, column)}}
+    else
+      _ ->
+        {:error, "Could not open source file: #{inspect(file_path)}"}
+    end
+  end
+
+  defp to_persisted_range(text, line, column) do
+    case Code.Fragment.surround_context(text, {line, column}) do
+      %{begin: begin, end: end_} ->
+        {begin_line, begin_column} = begin
+        {end_line, end_column} = end_
+        to_range({begin_line, begin_column}, {end_line, end_column})
+
+      _ ->
+        # NOTE: sometimes the column is 1, the cursor looks strange
+        # so we need to normalize it to the first char column
```

Comment:
<img width="384" alt="image" src="https://user-images.githubusercontent.com/12830256/229671504-18c48b49-fc69-4227-8393-5ddd260893e5.png">

I mean the above picture.

and after plus the leading_spaces, then become:

<img width="476" alt="image" src="https://user-images.githubusercontent.com/12830256/229671624-4cf3acc3-d5b1-4ef5-a8ac-96114a5af2f5.png">

Reply:
Can you update the comment so it describes the situation rather than saying it looks weird?
for example:

If the column is 1, but the code doesn't start on the first column, the cursor will be placed to the left of the actual definition, which isn't what we want.

Reply:
I prefer this:

> If the column is 1, but the code doesn't start on the first column, which isn't what we want. 
> The cursor will be placed to the left of the actual definition.